lib/puppet/provider/archive/ruby.rb

```diff
@@ -259,7 +259,9 @@ def gs_download(path)
   end
 
   def optional_switch(value, option)
-    if value
+    if value.is_a?(Array)
+      value.map { |item| option.map { |flags| flags % item } }.flatten
+    elsif value
       option.map { |flags| flags % value }
```

Comment:
How about this to eliminate some code duplication:

```suggestion
    if value
      Array(value).map { |item| option.map { |flags| flags % item } }.flatten
```

Tests pass locally with that.